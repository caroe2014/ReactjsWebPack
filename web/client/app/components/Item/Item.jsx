// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled from 'styled-components';
import { Flex, Text } from 'rebass';
import PropTypes from 'prop-types';
import Tile from 'web/client/app/components/Tile';
import { Trans } from 'react-i18next';
import { currencyLocaleFormat } from 'web/client/app/utils/NumberUtils';
import i18n from 'web/client/i18n';

const DetailsContainer = styled(Flex)`
  flex-direction: column;
  height: auto;
  flex: 1 1 auto;
  width: 100%;
`;

const DescriptionText = styled(Text)`
  color: ${props => props.theme.colors.secondaryTextColor};
  width: 100%;
  font-size: ${props => props.theme.fontSize.nm};
  margin: ${props => props.theme.spacing.sm} ${props => props.theme.spacing.none};
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const TopContainer = styled(Flex)`
  flex-direction: row;
`;

const TileContainer = styled(Flex)`
  flex-direction: row;
  margin: 0px 0px 8px 0px;
  ${props => props.theme.mediaBreakpoints.mobile`padding-left: 9px;`};
  overflow-x: auto;
  overflow-y: hidden;
  white-space: nowrap;
  flex-wrap: wrap;
  ${props => props.theme.mediaBreakpoints.mobile`display: none;`};
`;

const BottomContainer = styled(Flex)`
  width: 100%;
  justify-content: ${props => { return props.digitalMenuId ? `flex-end;` : `space-between`; }};
  display: flex;
  ${props => props.theme.mediaBreakpoints.desktop`border-top: 0.5px solid lightgrey;`};
  ${props => props.theme.mediaBreakpoints.tablet`
    border-top: 0.5px solid lightgrey; 
    padding-top: 14px;
  `};
  ${props => props.theme.mediaBreakpoints.mobile`display: none`};
  margin-bottom: 10px;
`;

const TileContainerNoDescription = styled(Flex)`
  flex-direction: row;
  ${props => props.theme.mediaBreakpoints.mobile`padding-left: 9px;`};
  overflow-x: auto;
  overflow-y: hidden;
  white-space: nowrap;
  flex-wrap: wrap;
  ${props => props.theme.mediaBreakpoints.mobile`display: none;`};
  margin-top: 10px;
`;

const MobileBottomContainer = styled(Flex)`
  width: 100%;
  justify-content: ${props => { return props.digitalMenuId ? `flex-end;` : `space-between`; }};
  display: flex;
  padding: 0px 10px;
  margin-bottom: 5px;
  margin-top: 5px;
  ${props => props.theme.mediaBreakpoints.desktop`display: none`};
  ${props => props.theme.mediaBreakpoints.tablet`display: none`};
  ${props => props.theme.mediaBreakpoints.mobile`display: flex`};
`;

const CartText = styled(Text)`
  color: ${props => props.theme.colors.buttonControlColor};
  font-size: ${props => props.theme.fontSize.nm};
  ${props => props.theme.mediaBreakpoints.desktop`margin-top: 14px;`};
  cursor: pointer;
  &:focus {
    outline: none;
    box-shadow: none;
  };
  order: 1;
`;

const AmountText = styled(Text)`
  color: ${props => props.theme.colors.primaryTextColor}; 
  font-size: ${props => props.theme.fontSize.nm};
  font-weight: bold;
  ${props => props.theme.mediaBreakpoints.desktop`margin-top: 14px;`};
  &:focus {
    outline: none;
    box-shadow: none;
  };
  order: 2
`;

const Tag = styled(Flex)`
  color: ${props => props.theme.colors.secondaryTextColor};
  font-size: 14px;
  display: inline-block;
  vertical-align: middle;
  margin-right: 2px;
  white-space: nowrap;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const fontStyle = {
  fontSize: '1em',
  display: 'inline-block',
  verticalAlign: 'middle',
  marginRight: '8px'
};

const IconTileContainer = styled(Flex)`
  flex-direction: row;
  height: 24px;
  justify-content: center;
  display: flex;
  align-items: center;
  ${props => props.theme.mediaBreakpoints.desktop`display: none;`};
  ${props => props.theme.mediaBreakpoints.tablet`display: none;`};
  ${props => props.theme.mediaBreakpoints.mobile`
  display: block;
  `};
`;

const IconTileBox = styled(Flex)`
  display: inline-block;
  vertical-align: middle;
  justify-contet:center;
  align-items: center;
  margin-right: ${props => {
    if (props.iscart) return '4px;';
    return '9px;';
  }};
  white-space: nowrap;
  border: ${props => {
    if (props.iscart) return 'none';
    return '1px solid #C5C5C5;';
  }};
  border-radius: ${props => {
    if (props.iscart) return 'none';
    return '16px;';
  }};
    padding: ${props => {
    if (props.iscart) return 'none';
    return '0px 4px 2px 4px;';
  }};
  
  ${props => props.theme.mediaBreakpoints.mobile`
    border: none;
    border-radius: none;
    padding: none;
    margin-right: 4px;
  `};
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const IconTileText = styled(Text)`
  color: ${props => props.theme.colors.secondaryTextColor};
  font-size: 14px;
  margin-top: 2px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const titleStyle = {
  fontSize: '16px',
  fontWeight: '500',
  wordBreak: 'break-word'
};

class Item extends Component {
  constructor (props) {
    super(props);
    this.onItemClick = this.onItemClick.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.canShowAttribute = this.canShowAttribute.bind(this);
  }
  onItemClick () {
    if (this.props.selectItem) {
      this.props.selectItem(this.props.keyProps);
    }
  }
  handleKeyDown (e) {
    // TODO: make this usable by all clickable elements,
    //  pass in caller's onclickhandler (13 is keycode for enter key, 32 is spacebar)
    if (e.which === 13 || e.which === 32) {
      this.onItemClick();
    }
  }

  canShowAttribute (itemDisplayList, itemProperty) {
    return itemDisplayList.length === 0 ? true : (itemDisplayList.find(itemDisplayParam => itemDisplayParam === itemProperty)); // eslint-disable-line max-len
  }

  render () {

    const { keyProps, showAddButton, addItemToCart, selectItem, iscart, ariaLabel, tabIndex,
      currencyDetails, digitalMenuId, itemDisplayList, ...rest } = this.props;

    const showDescription = this.canShowAttribute(itemDisplayList, 'description');
    const showLabels = this.canShowAttribute(itemDisplayList, 'labels');
    const showImage = this.canShowAttribute(itemDisplayList, 'image');
    const showPrice = this.canShowAttribute(itemDisplayList, 'price');

    const mobileFooter = showAddButton &&
      <MobileBottomContainer digitalMenuId={digitalMenuId}>
        {showPrice && keyProps.amount && <AmountText className='amount' tabIndex={0}>
          {currencyLocaleFormat(keyProps.amount, currencyDetails)}
        </AmountText>}
        {!digitalMenuId &&
          <CartText className='add-to-cart-text' onClick={this.onItemClick}
            aria-label={i18n.t('ADD_CART_TEXT')} role='button' tabIndex={0}>
            <Trans i18nKey='ITEM_ADD_TO_CART' />
          </CartText>
        }
      </MobileBottomContainer>;
    return (
      <Tile
        classContext={`tile item_${keyProps.id}`}
        {...rest}
        title={keyProps.displayText}
        image={showImage && keyProps.thumbnail}
        titleStyle={titleStyle}
        footer={mobileFooter}
        imageresponsive='true'
        iscart={iscart ? 1 : 0}
        ariaLabel={ariaLabel}
        tabIndex={tabIndex}
      >
        <DetailsContainer className='box-container' image={showImage && keyProps.image} >
          {showDescription && keyProps.description &&
            <TopContainer>
              <DescriptionText className='description'
                aria-label={`${i18n.t('ITEM_DESCRIPTION')}${keyProps.description}`} tabIndex={0}>
                {keyProps.description}</DescriptionText>
            </TopContainer>
          }
          <span>
            {showDescription && keyProps.description
              ? <TileContainer>
                {showLabels && keyProps.tagNames && keyProps.tagNames.map((tag) => (
                  <React.Fragment key={`itemTile_noImage_${tag}`}>
                    <IconTileBox iscart={iscart ? 1 : 0} className='icon-tile'
                      aria-label={tag}
                      role={tag.id !== 'calories' && 'img'}
                      tabIndex={0}>
                      {tag.id !== 'calories' &&
                        <i className={`agilysys-icon-${tag.toLowerCase().replace(/\s/g, '-')}`}
                          alt='' style={fontStyle}
                        />}
                      {!iscart
                        ? <Tag className='tag'>{tag}</Tag> : null}
                    </IconTileBox>
                  </React.Fragment>
                ))}
              </TileContainer>
              : <TileContainerNoDescription className='tile-container-no-description'>
                {showLabels && keyProps.tagNames && keyProps.tagNames.map((tag) => (
                  <React.Fragment key={`itemTile_noImage_${tag}`}>
                    <IconTileBox className='icon-tile' aria-label={tag}
                      role={tag.id !== 'calories' && 'img'}
                      tabIndex={0}>
                      {tag.id !== 'calories' &&
                        <i className={`agilysys-icon-${tag.toLowerCase().replace(/\s/g, '-')}`}
                          alt='' style={fontStyle} />}
                      {!iscart ? <Tag className='tag'>{tag}</Tag> : null}
                    </IconTileBox>
                  </React.Fragment>
                ))}
              </TileContainerNoDescription>
            }
            {showLabels && keyProps.tagNames && keyProps.tagNames.length > 0 && <IconTileContainer>
              {keyProps.tagNames.map((tag) => (
                <React.Fragment key={`itemTile-${tag}`} >
                  <IconTileBox className='icon-tile-mobile'
                    role={tag.id !== 'calories' && 'img'}
                    tabIndex={0} aria-label={tag.id !== 'calories' ? tag : tag.value}>
                    {tag.id === 'calories'
                      ? <IconTileText>{tag.value}</IconTileText>
                      : <i className={`agilysys-icon-${tag.toLowerCase().replace(/\s/g, '-')}`} style={fontStyle}
                        alt='' />
                    }
                  </IconTileBox>
                </React.Fragment>
              ))}
            </IconTileContainer>
            }
          </span>
        </DetailsContainer>
        {
          showAddButton &&
          <BottomContainer digitalMenuId={digitalMenuId}>
            {showPrice && keyProps.amount &&
              <AmountText
                className='amount'
                tabIndex={0}
                aria-label={`${i18n.t('ITEM_PRICE')}${currencyLocaleFormat(keyProps.amount, currencyDetails)}`}>
                {currencyLocaleFormat(keyProps.amount, currencyDetails)}
              </AmountText>}
            {!digitalMenuId && <CartText className='add-to-cart-text'
              onClick={this.onItemClick}
              onKeyDown={this.handleKeyDown}
              aria-label={i18n.t('ADD_CART_TEXT')} role='button' tabIndex={0}>
              <Trans i18nKey='ITEM_ADD_TO_CART' />
            </CartText>
            }
          </BottomContainer>
        }
      </Tile>
    );
  }
}

Item.propTypes = {
  keyProps: PropTypes.shape({
    id: PropTypes.string.isRequired,
    description: PropTypes.string,
    image: PropTypes.string,
    attributes: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired
      }))
  }).isRequired,
  currencyDetails: PropTypes.object
};

export default Item;
