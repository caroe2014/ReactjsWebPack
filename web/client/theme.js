// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
