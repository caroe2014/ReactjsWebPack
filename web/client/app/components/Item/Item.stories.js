// Copyright © 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import Item from './Item';
import { Provider } from 'rebass';
import theme from 'web/client/theme.js';

storiesOf('ItemComponent', module)
  .add('With image', () => {
    const menuItem = {
      id: 100,
      name: 'Mc Alister’s Club ',
      description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
      amount: '12.00',
      image: 'https://imagesvc.timeincapp.com/v3/mm/image?url=http%3A%2F%2Fcdn-image.myrecipes.com%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium_2x%2Fpublic%2F1426263343%2FBestEverJuicy-Burger_420x420.jpg%3Fitok%3DwNkrRxc4&w=700&q=85', // eslint-disable-line max-len
      tagNames: ['Gluten Free', 'Healthy Choice', 'Calories']
    };
    return (
      <Provider theme={theme}>
        <Item keyProps={menuItem} showAddButton/>
      </Provider>
    );
  }
  );

storiesOf('ItemComponent', module)
  .add('Without image', () => {
    const menuItem = {
      id: 100,
      name: 'Mc Alister’s Club ',
      description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
      amount: '12.00',
      attributes: [
        { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
        { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' }
      ]
    };
    return (
      <Provider theme={theme}>
        <Item keyProps={menuItem} showAddButton/>
      </Provider>
    );
  }
  );

storiesOf('ItemComponent', module)
  .add('With 2 line desc', () => {
    const menuItem = {
      id: 100,
      name: 'Mc Alister’s Club ',
      description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
      image: 'https://imagesvc.timeincapp.com/v3/mm/image?url=http%3A%2F%2Fcdn-image.myrecipes.com%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium_2x%2Fpublic%2F1426263343%2FBestEverJuicy-Burger_420x420.jpg%3Fitok%3DwNkrRxc4&w=700&q=85', // eslint-disable-line max-len
      amount: '12.00',
      attributes: [
        { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
        { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' }
      ]
    };
    return (
      <Provider theme={theme}>
        <Item keyProps={menuItem} showAddButton/>
      </Provider>
    );
  }
  );

storiesOf('ItemComponent', module)
  .add('With multi line desc', () => {
    const menuItem = {
      id: 100,
      name: 'Mc Alister’s Club ',
      description: 'Organic and natural sandwiches with delicious sauces made in all sizes. Sandwiches filled with nice cut veggies with slightly coated cheese.', // eslint-disable-line max-len
      image: 'https://imagesvc.timeincapp.com/v3/mm/image?url=http%3A%2F%2Fcdn-image.myrecipes.com%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium_2x%2Fpublic%2F1426263343%2FBestEverJuicy-Burger_420x420.jpg%3Fitok%3DwNkrRxc4&w=700&q=85', // eslint-disable-line max-len
      amount: '12.00',
      attributes: [
        { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
        { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
        { id: 'calories', name: 'Calories', value: '120Cal' }
      ]
    };
    return (
      <Provider theme={theme}>
        <Item keyProps={menuItem} showAddButton/>
      </Provider>
    );
  }
  );

storiesOf('ItemComponent', module)
  .add('Without amount', () => {
    const menuItem = {
      id: 100,
      name: 'Mc Alister’s Club ',
      description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
      image: 'https://imagesvc.timeincapp.com/v3/mm/image?url=http%3A%2F%2Fcdn-image.myrecipes.com%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium_2x%2Fpublic%2F1426263343%2FBestEverJuicy-Burger_420x420.jpg%3Fitok%3DwNkrRxc4&w=700&q=85', // eslint-disable-line max-len
      attributes: [
        { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
        { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
        { id: 'calories', name: 'Calories', value: '120Cal' }
      ]
    };
    return (
      <Provider theme={theme}>
        <Item keyProps={menuItem} showAddButton/>
      </Provider>
    );
  }
  );

storiesOf('ItemComponent', module)
  .add('With 1 tile', () => {
    const menuItem = {
      id: 100,
      name: 'Mc Alister’s Club ',
      description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
      image: 'https://imagesvc.timeincapp.com/v3/mm/image?url=http%3A%2F%2Fcdn-image.myrecipes.com%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium_2x%2Fpublic%2F1426263343%2FBestEverJuicy-Burger_420x420.jpg%3Fitok%3DwNkrRxc4&w=700&q=85', // eslint-disable-line max-len
      amount: '12.00',
      attributes: [
        { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' }
      ]
    };
    return (
      <Provider theme={theme}>
        <Item keyProps={menuItem} showAddButton/>
      </Provider>
    );
  }
  );

storiesOf('ItemComponent', module)
  .add('With no tiles', () => {
    const menuItem = {
      id: 100,
      name: 'Mc Alister’s Club ',
      description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
      image: 'https://imagesvc.timeincapp.com/v3/mm/image?url=http%3A%2F%2Fcdn-image.myrecipes.com%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium_2x%2Fpublic%2F1426263343%2FBestEverJuicy-Burger_420x420.jpg%3Fitok%3DwNkrRxc4&w=700&q=85', // eslint-disable-line max-len
      amount: '12.00'
    };
    return (
      <Provider theme={theme}>
        <Item keyProps={menuItem} showAddButton/>
      </Provider>
    );
  }
  );
