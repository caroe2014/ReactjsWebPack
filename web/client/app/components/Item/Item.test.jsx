// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';
import { t } from 'i18next';
import Item from './Item';

configure({ adapter: new Adapter() });

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

jest.mock('i18next');
t.mockImplementation((key) => {
  return key;
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('ItemComponent', () => {

  let wrapper;
  let mockProps = {
    id: '100',
    name: 'This is a mock name',
    description: 'Mock description text',
    amount: 123,
    thumbnail: 'image.png',    
    tagNames: [
      'mockName1',
      'mockName2',
      'mockName3'
    ]
  };

  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Item keyProps={mockProps} itemDisplayList={['labels', 'description', 'image', 'price']} showAddButton/>
    );
  });

  //TODO: uncomment
  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    expect(mount(<ThemeProvider theme={theme}><Item keyProps={mockProps} itemDisplayList={['labels', 'description', 'image']} showAddButton/></ThemeProvider>).exists()).toBe(true);
  });

  it('should render Image when Image prop is received', () => {
    expect(wrapper.dive().find('ImageContainer').length).toBe(1);
  });

  describe('when image prop is not received and image is not present in itemDisplayList prop', () => {
    beforeEach(() => {
      mockProps.thumbnail = '';
    });
    it('should not render Image when Image prop is received', () => {
      const imageWrapper = shallow(<Item keyProps={mockProps} itemDisplayList={['labels']}/>);
      expect(imageWrapper.dive().find('ImageContainer').length).toBe(0);
    });
  });

  describe('when image is not present in itemDisplayList prop and image prop is received', () => {
    it('should not render Image', () => {
      const imageWrapper = shallow(<Item keyProps={mockProps} itemDisplayList={['labels']}/>);
      expect(imageWrapper.dive().find('ImageContainer').length).toBe(0);
    });
  });

  describe('when tagNames prop is received', () => {
    it('should render all attribute tiles', () => {
      expect(wrapper.find('.icon-tile').length).toBe(mockProps.tagNames.length);
    });
  });

  describe('when tagNames prop is received', () => {
    it('should render all Icon tiles', () => {
      expect(wrapper.find('.icon-tile-mobile').length).toBe(mockProps.tagNames.length);
    });
  });

  describe('when tagNames prop is not received and labels is not present in itemDisplayList prop', () => {
    beforeEach(() => {
      mockProps.tagNames = [];
    });
    it('should not render attributes tiles', () => {
      const tileWrapper = shallow(<Item keyProps={mockProps} itemDisplayList={['image']} iscart/>);
      expect(tileWrapper.find('.icon-tile').length).toBe(0);
    });
  });

  describe('when labels is not present in itemDisplayList prop and tagNames prop is received', () => {
    it('should not render attributes tiles', () => {
      const tileWrapper = shallow(<Item keyProps={mockProps} itemDisplayList={['image']} iscart/>);
      expect(tileWrapper.find('.icon-tile').length).toBe(0);
    });
  });

  describe('when amount prop is received and price is present in itemDisplayList prop', () => {
    it('should render amount text', () => {
      expect(wrapper.find('.amount').length).toBe(1);
    });
  });

  describe('when amount prop is not received', () => {
    beforeEach(() => {
      mockProps.amount = '';
    });
    it('should not render amount text', () => {
      const amountWrapper = shallow(<Item keyProps={mockProps} itemDisplayList={['image']} showAddButton/>);
      expect(amountWrapper.find('.amount').length).toBe(0);
    });
  });

  describe('when description is not available', () => {
    const newMockProps = {
      id: '100',
      name: 'This is a mock name',
      amount: 'Mock amount',
      thumbnail: 'image.png',
      tagNames: [
        'mockName1',
        'mockName2',
        'mockName3'
      ]
    };
    it('should render TileContainerNoDescription when description is present in itemDisplayList prop', () => {
      const shallowWrapper = shallow(<Item keyProps={newMockProps} itemDisplayList={['description']} showAddButton/>);
      expect(shallowWrapper.find('.tile-container-no-description').length).toBe(1);
    });
  });

});
