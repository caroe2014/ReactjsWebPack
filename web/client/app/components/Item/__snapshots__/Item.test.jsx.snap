// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemComponent does a snapshot test 1`] = `
<Tile
  classContext="tile item_100"
  footer={
    <Styled(Flex)>
      <Styled(Text)
        className="amount"
        tabIndex={0}
      >
        $123.00
      </Styled(Text)>
      <Styled(Text)
        aria-label="ADD_CART_TEXT"
        className="add-to-cart-text"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          i18nKey="ITEM_ADD_TO_CART"
        />
      </Styled(Text)>
    </Styled(Flex)>
  }
  hasBorder={true}
  image="image.png"
  imageresponsive="true"
  isEnabled={true}
  iscart={0}
  titleStyle={
    Object {
      "fontSize": "16px",
      "fontWeight": "500",
      "wordBreak": "break-word",
    }
  }
>
  <Styled(Flex)
    className="box-container"
  >
    <Styled(Flex)>
      <Styled(Text)
        aria-label="ITEM_DESCRIPTIONMock description text"
        className="description"
        tabIndex={0}
      >
        Mock description text
      </Styled(Text)>
    </Styled(Flex)>
    <span>
      <Styled(Flex)>
        <Styled(Flex)
          aria-label="mockName1"
          className="icon-tile"
          iscart={0}
          role="img"
          tabIndex={0}
        >
          <i
            alt=""
            className="agilysys-icon-mockname1"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "1em",
                "marginRight": "8px",
                "verticalAlign": "middle",
              }
            }
          />
          <Styled(Flex)
            className="tag"
          >
            mockName1
          </Styled(Flex)>
        </Styled(Flex)>
        <Styled(Flex)
          aria-label="mockName2"
          className="icon-tile"
          iscart={0}
          role="img"
          tabIndex={0}
        >
          <i
            alt=""
            className="agilysys-icon-mockname2"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "1em",
                "marginRight": "8px",
                "verticalAlign": "middle",
              }
            }
          />
          <Styled(Flex)
            className="tag"
          >
            mockName2
          </Styled(Flex)>
        </Styled(Flex)>
        <Styled(Flex)
          aria-label="mockName3"
          className="icon-tile"
          iscart={0}
          role="img"
          tabIndex={0}
        >
          <i
            alt=""
            className="agilysys-icon-mockname3"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "1em",
                "marginRight": "8px",
                "verticalAlign": "middle",
              }
            }
          />
          <Styled(Flex)
            className="tag"
          >
            mockName3
          </Styled(Flex)>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)>
        <Styled(Flex)
          aria-label="mockName1"
          className="icon-tile-mobile"
          role="img"
          tabIndex={0}
        >
          <i
            alt=""
            className="agilysys-icon-mockname1"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "1em",
                "marginRight": "8px",
                "verticalAlign": "middle",
              }
            }
          />
        </Styled(Flex)>
        <Styled(Flex)
          aria-label="mockName2"
          className="icon-tile-mobile"
          role="img"
          tabIndex={0}
        >
          <i
            alt=""
            className="agilysys-icon-mockname2"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "1em",
                "marginRight": "8px",
                "verticalAlign": "middle",
              }
            }
          />
        </Styled(Flex)>
        <Styled(Flex)
          aria-label="mockName3"
          className="icon-tile-mobile"
          role="img"
          tabIndex={0}
        >
          <i
            alt=""
            className="agilysys-icon-mockname3"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "1em",
                "marginRight": "8px",
                "verticalAlign": "middle",
              }
            }
          />
        </Styled(Flex)>
      </Styled(Flex)>
    </span>
  </Styled(Flex)>
  <Styled(Flex)>
    <Styled(Text)
      aria-label="ITEM_PRICE$123.00"
      className="amount"
      tabIndex={0}
    >
      $123.00
    </Styled(Text)>
    <Styled(Text)
      aria-label="ADD_CART_TEXT"
      className="add-to-cart-text"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        i18nKey="ITEM_ADD_TO_CART"
      />
    </Styled(Text)>
  </Styled(Flex)>
</Tile>
`;
